timescaledb/how-to-guides/compression/backfill-historical-data.md

```diff
@@ -1,97 +1,104 @@
+You can backfill data by using the `decompress_backfill` function. Before you
+begin, add the [`decompress_backfill` function and its supporting
+functions][timescaledb-extras-backfill] to your database.
 
-Similar to above, you can also decompress a set of chunks based on a
-time range by first looking up this set of chunks via `show_chunks`:
+<procedure>
 
-``` sql
-SELECT decompress_chunk(i) from show_chunks('conditions', newer_than, older_than) i;
-```
+## Backfilling with the backfilling function
 
-<highlight type="tip">
-You need to run 'decompress_chunk' for each chunk that is impacted
-by your INSERT or UPDATE statement in backfilling data. Once your needed chunks
-are decompressed you can proceed with your data backfill operations.
+<highlight type="note">
+Backfill data by first inserting it into a temporary table. Temporary tables are
+short-lived. They are automatically dropped when the database session ends. No
+clean-up is required. If you backfill regularly, you can choose to use a normal
+table instead. This allows multiple writers to insert data into the table at the
+same time. When backfilling is complete, clean up by truncating or dropping your
+table.
 </highlight>
 
-Once your backfill and update operations are complete we can simply re-enable
-our compression policy job:
-
-``` sql
-SELECT alter_job(<job_id>, scheduled => true);
-```
-
-This job re-compresses any chunks that were decompressed during your backfilling
-operation the next time it runs. To have it run immediately, you can expressly execute
-the command via [`run_job`][run-job]:
-
-``` sql
-CALL run_job(<job_id>);
-```
+1.  At the `psql` prompt, create a table with the same schema as the hypertable
+    you're backfilling into. In this example, the hypertable is named `cpu`.
+    ```sql
+    CREATE TEMPORARY TABLE cpu_temp AS SELECT * FROM cpu WITH NO DATA;
+    ```
+
+1.  Insert your data into the backfill table.
+
+1.  Call the supplied backfill function. This function:
+      * Halts the compression policy
+      * Identifies the chunks where the backfilled data belongs
+      * Decompresses the chunks
+      * Inserts data from the temporary table into the hypertable
+      * Re-enables the compression policy
+    ```sql
+    CALL decompress_backfill(staging_table=>'cpu_temp', destination_hypertable=>'cpu');
+    ```
+
+</procedure>
+
+## Backfill manually
+If you prefer not to use the `decompress_backfill` function, you can perform its
+steps manually. In this procedure, you turn off your compression policy and
+identify your compressed chunks before inserting your backfilled data.
+
+<procedure>
+
+### Backfilling manually
+1.  At the `psql` prompt, find the `job_id` of the compression policy:
+    ```sql
+    SELECT s.job_id
+    FROM timescaledb_information.jobs j
+      INNER JOIN timescaledb_information.job_stats s ON j.job_id = s.job_id
+      WHERE j.proc_name = 'policy_compression' AND s.hypertable_name = <TARGET_TABLE>;
```

Comment:
```suggestion
      WHERE j.proc_name = 'policy_compression' AND s.hypertable_name = <TARGET_TABLE>;
```
I'd really like to get some input here from the larger team. I'm tired enough tonight that I honestly cannot think of what I'd want, but this is one of the few places I recall seeing the tag-like placeholders for variable or parameter names, and I just don't think it works.

In the old docs we appear to have mixed things, which isn't better, but in context we tend to use sample table/parameter values. I understand it's not "that much" clearer, but the correct way to substitute parameters in any language is always tricky IMO.

Maybe a cursory  look at what some other similar docs/SQL do?

Reply:
Thanks for the input, Ryan. I did a cursory review of what other industry documentation uses:

|Docs|Formal-ish syntax definitions|Sample values|Tag-like placeholders|Notes|
|-|-|-|-|-|
|PostgreSQL|✅|✅||PostgreSQL is not the best example to learn from, because their docs don't have how-to guides that I've found. They're mainly API references, which start with a formal-ish syntax definition and end with examples. This is a good pattern for our references but not as useful in context for a task or concept.|
|InfluxDB||✅|✅|Uses a tag-like placeholder immediately followed by an example|
|Stripe|✅||✅|Leans towards sample values in their how-to guides. Placeholders are rare.|
|Clickhouse||✅|||
|Snowflake||✅|✅|Leans towards tag-like placeholders|
|MongoDB||✅||Uses example variables but heavily defines their examples, like literally "Imagine a pizza store receives two orders..."|

In our recent discussion on this topic, we decided on placeholder parameters as the primary method with examples when needed, but maybe we should flip that, given sample variables seem to be more popular? 

I do think some examples need to be better defined, for example if we were to replace `<JOB_ID>` below with the actual job id, it wouldn't be clear at a quick glance what that string of numbers is supposed to be, so it would require more explanatory text

Reply:
I probably am leaning towards the Influx method, but it's going to require more words generally, and a concerted effort to go through and update everything.

Reply:
I like the Snowflake approach only because we may not have to do a lot of rewrites.